### PR TITLE
[W4.3d][T16-A2]KENNARD NG POOL HUA

### PR DESCRIPTION
**Pull request by Kennard123661**
Segment Address into Block, PostalCode, Street and Unit classes.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -26,7 +40,29 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+            postalCode = new PostalCode(addressSegments[POSTAL_CODE_INDEX].trim());
+        }
+
+        if(numAddressSegments > STREET_INDEX) {
+            street = new Street(addressSegments[STREET_INDEX].trim());
+        }
+
+        if(numAddressSegments > UNIT_INDEX) {
+            unit = new Unit(addressSegments[UNIT_INDEX].trim());
+        }
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Missing header comment. All non trivial methods should have javadoc format header comments.
This code chunk is functionally correct, however is there another way of making this more readable =)?

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -38,19 +74,37 @@ public static boolean isValidAddress(String test) {
...
+
+        if(unit != null) {
+            address += unit.toString() + ", ";
+        }
+
+        if(postalCode != null) {
+            address += postalCode.toString() + ", ";
+        }
+
+        return address.substring(0, address.length() - 2);
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/Address.java]: Would refactoring out `", "` in this case improve code readability?

File: src/seedu/addressbook/data/person/address/Block.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package seedu.addressbook.data.person.address;
+
+public class Block {
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/data/person/address/Block.java]: Missing header comment. There should be header comments to describe the purpose of the class.

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (Kennard123661) [reply src/seedu/addressbook/data/person/Address.java]: Hi @brandonyeoxg , my guess is to extract it out as a method, for example, hasBlock(), which returns true if numAddressSegments > UNIT_INDEX

**Pull request closed without merging**